src/components/CustomRegisterField/CustomRegisterField.tsx

```diff
@@ -0,0 +1,77 @@
+import { type FC, type ChangeEventHandler, type ReactNode, useMemo, useCallback } from 'react';
+import type { GetRegisterFieldOption } from '@inplayer-org/inplayer.js';
+
+import Checkbox from '#components/Checkbox/Checkbox';
+import TextField from '#components/TextField/TextField';
+import Radio from '#components/Radio/Radio';
+import Dropdown from '#components/Dropdown/Dropdown';
+import DateField from '#components/DateField/DateField';
+import { ConsentFieldVariants, REGISTER_FIELD_VARIANT } from '#src/services/inplayer.account.service';
+import { countries, usStates } from '#static/json';
+
+type Props = {
+  type: ConsentFieldVariants;
+  name: string;
+  value: string | boolean;
+  onChange: (name: string, value: string | boolean) => void;
+} & Partial<{
+  label: ReactNode;
+  placeholder: string;
+  error: boolean;
+  helperText: string;
+  disabled: boolean;
+  required: boolean;
+  options: GetRegisterFieldOption;
+}>;
+
+export type CustomRegisterFieldCommonProps = Props;
+
+export const CustomRegisterField: FC<Props> = ({ type, name, value = '', onChange, ...props }) => {
+  const optionsList = useMemo(() => {
+    const optionsObject = (() => {
+      switch (type) {
+        case REGISTER_FIELD_VARIANT.COUNTRY_SELECT:
+          return countries;
+        case REGISTER_FIELD_VARIANT.US_STATE_SELECT:
+          return usStates;
+        default:
+          return props.options;
+      }
+    })();
+
+    if (!optionsObject) {
+      return [];
+    }
+
+    return Object.entries(optionsObject).map(([value, label]) => ({ value, label }));
+  }, [type, props.options]);
+
+  const changeHandler: ChangeEventHandler<HTMLInputElement> = useCallback(
+    ({ currentTarget }) => {
+      const value = type === REGISTER_FIELD_VARIANT.CHECKBOX ? (currentTarget as HTMLInputElement).checked : currentTarget.value;
```

Comment:
this is also handled in the parent `onChange` method. Can we just pass the values to the parent's `onChange`?

Reply:
As I already answered, the difference is that the parent's `onChange` method requires an event object as its only argument, while the method with the same name here requires two arguments `name` and `value`, and the only reason for that is the `DateField` component which doesn't produce an event object.